File: src/leetcode/python/TwoSum_1.py
import collections


class TwoSum_1(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        index = {}
        for i in range(len(nums)):
            k = target - nums[i]
            if nums[i] in index:
                return [index[nums[i]], i]
            else:
                index[k] = i

File: src/leetcode/java/TwoSum_1.java
package leetcode.java;

import java.util.HashMap;

public class TwoSum_1 {
  public int[] twoSum(int[] nums, int target) {
    int[] res = new int[2];
    HashMap<Integer, Integer> index = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      int key = target - nums[i];
      if (index.containsKey(nums[i])) {
        res[0] = index.get(nums[i]);
        res[1] = i;
      } else {
        index.put(key, i);
      }
    }
    return res;
  }
}
